python/looker_sdk/rtl/api_settings.py

```diff
@@ -32,13 +32,21 @@
 from looker_sdk.rtl import transport
 
 if sys.version_info >= (3, 8):
-    from typing import Protocol
+    from typing import Protocol, TypedDict
 else:
-    from typing_extensions import Protocol
+    from typing_extensions import Protocol, TypedDict
 
+from typing_extensions import Required
+
+class SettingsConfig(TypedDict, total=False):
+    client_id: Required[str]
+    client_secret: Required[str]
+    base_url: Required[str]
```

Comment:
`base_url` is technically not required. It's required to be an instance property on the settings object and the default implementation gets it from `read_config` but that's not required. I know it's confusing - honestly the only reason `read_config` exists is so that our default implementation doesn't have to store `client_id` or `client_secret` in python memory for security purposes. Otherwise they'd just be required instance properties too and there would be no `read_config`